### PR TITLE
Add max length value for string fields of entity

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
The existing implementation doesn't allow to setting the length of string field in the entity. This pr provides a way to set the length of the field by using constraint package's string annotation.

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/3313

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistGenerateSqlScript.java [hunk 1]
```diff
@@ -93,15 +93,18 @@ private static String generateFieldsQuery(RecordTypeDescriptorNode recordNode, S
         Optional<MetadataNode> metadata;
         String fieldName;
         String sqlScript = EMPTY;
+        String length = "191";
```
Comment (daneshk) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistGenerateSqlScript.java:96]: Let's make this number constant

File: compiler-plugin-test/src/test/resources/test-src/sql_script/package_10/sample.bal [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    Item1 item1?;
+};
+
+@persist:Entity {key: ["id"]}
+public type Item1 record {
+    @persist:AutoIncrement
+    readonly int id = 5;
+    @constraint:String {
+        maxLength: 20
+    }
```
Comment (daneshk) [inline compiler-plugin-test/src/test/resources/test-src/sql_script/package_10/sample.bal:58]: is it only applicable to string type in SQL?

**Comment by daneshk** [comment]
@MadhukaHarith92 @kaneeldias Better we add the constraint validate in the client create/update functions. WDYT?

**Comment by daneshk** [comment]
> @MadhukaHarith92 @kaneeldias Better we add the constraint validate in the client create/update functions. WDYT?

Created an issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/3360

**Comment by MadhukaHarith92** [comment]
> @MadhukaHarith92 @kaneeldias Better we add the constraint validate in the client create/update functions. WDYT?

Yes, +1

File: compiler-plugin-test/src/test/resources/test-src/sql_script/package_10/sample.bal (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply compiler-plugin-test/src/test/resources/test-src/sql_script/package_10/sample.bal:58]: The existing impl supports other sql type such as int, float and decimal. Those are supported to set the length but it seems those purpose are different with  the constraint package's max value purpose. 

As per the MYSQL doc, https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html

- INT(M) indicates the maximum display width.
     Eg:  If we define the int type fields like this, 
     > CREATE TABLE Products(
     >        product_id INT,
     >        weight int(2) ZEROFILL,
     >        review int(3) ZEROFILL,
     >        PRIMARY KEY (product_id)
     >  );
     > 
     > INSERT INTO Products (product_id, weight, review) VALUES (5, 5, 5);
     > SELECT * FROM Products;

    the output is:
    > product_id	weight	review
    > 5	                 05	         005

- FLOAT(M,D) or REAL(M,D) or DOUBLE PRECISION(M,D) or DECIMAL(M,D). Here, (M,D) means than values can be stored with up to M digits in total, of which D digits may be after the decimal point.
  Eg:  If we define the float type fields like this, 
   > CREATE TABLE demo (
   >          f1 FLOAT(2, 1),
   >          f2 FLOAT(4, 3),
   >          f3 FLOAT(7, 5)
   >      );
   > 
   >  INSERT INTO demo(f1,f2,f3)VALUES(1 / 3, 1 / 3, 1 / 3);
   >     
   >  SELECT * FROM demo;

   the output is:
   >  f1	f2	            f3
   >  0.3.  0.333	0.33333

**Review by daneshk — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**